bin/start_sync_stop.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python
+"""A script to test rippled in an infinite loop of start-sync-stop.
+
+- Expects 'rippled' to be on the `PATH` and 'rippled.cfg' to be in the current
+  directory.
+- Takes no arguments.
+- Can be stopped with SIGINT.
+- Tested on Python 3.8.
+- Has no dependencies outside the standard library.
+"""
+
+import asyncio
+import contextlib
+import json
+import logging
+import os
+import subprocess
+import time
+import urllib.error
+import urllib.request
+
+
+@contextlib.asynccontextmanager
+async def rippled(path='rippled', config='rippled.cfg'):
+    """A context manager for a rippled process."""
+    # Start the server.
+    process = await asyncio.create_subprocess_exec(
+        path,
+        '--conf',
+        config,
+        stdout=subprocess.DEVNULL,
+        stderr=subprocess.DEVNULL,
+    )
+    logging.info(f'rippled started with pid {process.pid}')
+    try:
+        yield process
+    finally:
+        # Ask it to stop.
+        logging.info(f'asking rippled ({process.pid}) to stop')
+        start = time.time()
+        process.terminate()
+
+        # Wait nicely.
+        try:
+            await asyncio.wait_for(process.wait(), 60)
+        except asyncio.TimeoutError:
+            # Ask the operating system to kill it.
+            logging.warning(f'killing rippled ({process.pid})')
+            try:
+                process.kill()
+            except ProcessLookupError:
+                pass
+
+        code = await process.wait()
+        end = time.time()
+        logging.info(
+            f'rippled stopped after {end - start} seconds with code {code}'
```

Comment:
This echos a _lot_ of significant digits.  Consider reducing the visual noise with something like...
```
        logging.info(
            'rippled stopped after {0:.2f} seconds with code {1}'.format(
                end - start, code)
```